Tasks/NuGetInstaller/nugetinstaller.ts

```diff
@@ -47,62 +52,110 @@ var runnuget = function(fn) {
     })
 }
 
-// Resolve files for the specified value or pattern
-var filesList : string [];
-if (solution.indexOf('*') == -1 && solution.indexOf('?') == -1) {
-    // No pattern found, check literal path to a single file
-    tl.checkPath(solution, 'files');
+// get auth token and set credentials to NuGet config file
+var tempNugetConfigPath = '';
+if (nugetConfigPath && nugetConfigPath.toLowerCase() != sourcesFolder.toLowerCase()) {
 
-    // Use the specified single file
-    filesList = [solution];
+    tl._writeLine('Get end point authorization');
+    var auth = tl.getEndpointAuthorization('SYSTEMVSSCONNECTION', false);
+    if (auth.scheme == 'OAuth') {
+        var accessToken = auth.parameters['AccessToken'];
+        tl.debug('accessToken: ' + accessToken);
+    }
+    else {
+        tl.warning('Could not determine credentials to use for NuGet');
+    }
 
-} else {
-    var firstWildcardIndex = function(str) {
-        var idx = str.indexOf('*');
+    var nugetHelper = new helper.VsoNuGetHelper(nuGetPathToUse, nugetConfigPath, accessToken);
```

Comment:
You might want to use buildId to make the file name unique to a build. Consider a test cases where multiple nuget tasks are added to a build as well. Do you wish to have different temp files for each?